File: Clase 16/PYTHON/ejemplo_animales/Animal.py
class Animal:

    def __init__(self, id, nombre, especie):
        self.id = id
        self.nombre = nombre
        self.especie = especie

    def se_mueve(self):
        print(f"{self.nombre} se mueve...")

    def hace_sonido(self):
        print(f"{self.nombre} hace un sonido...")

File: Clase 18/JAVA/ejemploAnimales/Animal.java
package ejemploAnimales;

public class Animal{
    protected int id;
    protected String nombre;
    protected String especie;

    public Animal(int id, String nombre, String especie){
        this.id = id;
        this.nombre = nombre;
        this.especie = especie;
    }

    // Polimorfismo Parametrico
    public Animal(int id){
        this.id = id;
        this.nombre = "";
        this.especie = "";
    }

    public void seMueve(){
        if(this.nombre == "")
            System.out.println("Animal X se esta moviendo");
        else
            System.out.println(this.nombre + " se esta moviendo");
    }

    public void haceSonido(){
        if(this.nombre == "")
            System.out.println("Animal X hace un sonido...");
        else
            System.out.println(this.nombre + " hace un sonido...");
    }
}